log4j-api-java9/src/main/java/org/apache/logging/log4j/util/StackLocator.java

```diff
@@ -31,6 +32,11 @@
 
     private final static StackLocator INSTANCE = new StackLocator();
 
+    private final static ThreadLocal<CallerLocator> LOCATOR = new ThreadLocal<CallerLocator>() {
+        public CallerLocator initialValue() {
+            return new CallerLocator();
+        }
+    };
```

Comment:
Since this is java 9, we have the option to take advantage of the java 8`ThreadLocal.withInitial(CallerLocator::new)`